### DISCUSSION TITLE
Is there a way to render a "time since timestamp" in days as a field?

### DISCUSSION THREAD

**Discussion by ChaosNicro** [gsantner/markor#2192] [Q&A] [answered]
Heya,
For my tasks, one of the things I have to track are "refill"-tasks. Stuff to periodically check roughly X days after you last checked them. I don't want to use a calender-event for them, because it's not a "do on this day"-thing and the next event would depend on when I actually get to it.

Theoretically, the only data to store is the timestamp of the last check, which combined with the current time yields the delta needed. I've seen people use JS magic for such views and wanted to ask if someone knew a script to get: "Slug: X days since dd.MM.yyyy" for every line in a file, where X is dynamic and I could update the timestamps as needed.
Thanks in advance.

**Comment by harshad1** [comment]
What you need is this:

https://updatenotes.blog/todotxt-recurring-tasks/

We don't support this in markor (yet). 

Though since we work on plain offline files a termux script could do this for you

**Reply by ChaosNicro** [reply]
This somewhat fits, but I would like to solve for the delta not the next due date in the equation. 
Rec-Tasks do: "stored completion date + fixed delta = dynamic next due date"
I would like: " dynamic current date - stored last completion date = dynamic delta"
I'm sure I can abuse the JS parser for this, but I haven't gotten around to it. Termux doesn't really do, because I want this to be refreshed when I open the list.

**Comment by ChaosNicro** [comment, marked as answer]
With a bit of AI-help, I hacked together this snippet, only drawback is that timestamps must be supplied in generic ISO and dates might be of by one because they account for time. Didn't fix, because JS has terrible date parsing and I don't know if API like time or moment work in Markor.
```
<script>
  // Define your list of timestamps and titles
  const timestampList = [
    { title: "title", timestamp: "2023-12-16" },
    { title: "title2", timestamp: "2023-11-19" },
    { title: "title3", timestamp: "2023-12-30" },
    { title: "title4", timestamp: "2023-11-28" },
    // Add more events as needed
  ];

  // Function to calculate the days between two dates
  function calculateDaysDifference(date1, date2) {
    const oneDay = 24 * 60 * 60 * 1000; // hours*minutes*seconds*milliseconds
    const diffDays = Math.round(Math.abs((date1 - date2) / oneDay));
    return diffDays;
  }

  // Function to format the output
  function formatTimestamps() {
    const currentDate = new Date();

    // Iterate through the timestampList and output the formatted string
    timestampList.forEach((event) => {
      const eventDate = new Date(event.timestamp);
      const daysDifference = calculateDaysDifference(currentDate, eventDate);
      const output = `${event.title}: ${daysDifference} days since ${eventDate.toLocaleDateString()}`;
      document.write(output + "<br>");
    });
  }

  // Call the function when the page loads
  formatTimestamps();
</script>
```
